### PR TITLE
Update nixpkgs and GHCs for cabal / stack, test with PG 17

### PR DESCRIPTION
**Pull request by wolfgangwalther**
Update and align everything. Let's see which parts pass and which don't.

### CODE REVIEW

File: test/spec/fixtures/schema.sql [hunk 1]
```diff
@@ -257,7 +257,7 @@ CREATE FUNCTION varied_arguments(
   enum enum_menagerie_type,
   arr text[],
   "integer" integer default 42,
-  json json default '{}',
+  "json" json default '{}',
```
Comment (steve-chavez) [inline test/spec/fixtures/schema.sql:260]: So pg 17 finally fails on these identifiers? Or is it just a cleanup?

**Comment by steve-chavez** [comment]
Q: I thought I saw GHC 9.10.1 in this PR before, why was it removed? Something broke?

I was hoping to do some experiments with Linear Types, GHC 9.10.1 added support for using it with let/where ([ref](https://discourse.haskell.org/t/ghc-9-10-1-is-now-available/9523)).

**Comment by steve-chavez** [comment]
I think we discussed this before, but couldn't we just delete the `cabal.project.freeze`? I don't recall why we needed it if we have Nix.

**Comment by develop7** [comment]
> I *think* this would silently pull in new dependencies for cabal-based builds, which we still do. This makes CI not reproducible anymore. Not sure whether that was the original reason why it was added, though.

It's exactly the reason I've introduced it in https://github.com/PostgREST/postgrest/pull/3004. Should've added a comment there too, though.

Also I wonder if I could fix these more explicitly by introducing upper bound in `cabal.project`

**Pull request merged**